ykrt/src/thread_intercept.rs

```diff
@@ -1,11 +1,59 @@
 use std::os::raw::{c_int, c_void};
 
+use libc::{dlsym, free, malloc, pthread_create};
+use std::{ffi::CString, ptr::null_mut};
+
+#[derive(Debug)]
+struct Target {
+    pub func: extern "C" fn(*mut c_void) -> *mut c_void,
+    pub arg: *mut c_void,
+}
+
+extern "C" fn wrap_thread_routine(tgt: *mut c_void) -> *mut c_void {
+    let str = CString::new("shadowstack_0").unwrap();
+    let tgt = unsafe { Box::from_raw(tgt as *mut Target) };
+    // Obtain address of a shadowstack_0 symbol
+    let shadowstack_symbol_addr = unsafe { dlsym(null_mut(), str.as_ptr()) };
+    if shadowstack_symbol_addr.is_null() {
+        panic!("Unable to find shadowstack address")
+    }
+    let newsstack = unsafe { malloc(10000) };
```

Comment:
shall we define the new thread's shadow stack size with a const somewhere?

Reply:
Ah yes, and we should probably make it the same size as the shadowstack in main.

Reply:
[489f989](https://github.com/ykjit/yk/pull/1664/commits/489f9892308ea8b0114ebd1264ea19d3dc086d00)